### PR TITLE
Optimize gradle & Remove redundant .gitignore files

### PR DESCRIPTION
**Pull request by Goooler**
- Add `base.gradle` & `kotlin.gradle` for modules to apply and simplifying boilerplate.
- Remove redundant .gitignore files.
- Extrace version strings to ext block.
- Add arg to [Optimizing Gradle builds for JDK 11](https://developer.android.com/studio/releases/gradle-plugin#optimize-gc-jdk-11).
- Add [Dependabot](https://dependabot.com/) to automated dependency updates.

### CODE REVIEW

File: .github/dependabot.yml [hunk 1]
```diff
@@ -0,0 +1,7 @@
+version: 2
```
Comment (gpeal) [inline .github/dependabot.yml:1]: I would rather not add dependabot. It will actually add more work for me to manage the pull requests as a single maintainer.

File: gradle/base.gradle [hunk 2]
```diff
@@ -0,0 +1,18 @@
+android {
```
Comment (gpeal) [inline gradle/base.gradle:1]: Can you undo this change? I'd rather keep the build files self-contained. There aren't that many modules and I would like each one, especially the ones that get published, to be very explicit.

Comment (Goooler) [reply gradle/base.gradle:1]: I can revert them. 
But I think I just extracted common parts of modules' gradle, and properties could be overrided.

File: .github/dependabot.yml (hunk 1 above)
Comment (Goooler) [reply .github/dependabot.yml]: Removed.

File: build.gradle [hunk 3]
```diff
@@ -2,9 +2,20 @@ import org.ajoberstar.grgit.Grgit
...
+    lifecycleVersion = '2.3.1'
+    composeVersion = '1.0.2'
     kotlinVersion = '1.5.21'
     daggerVersion = '2.38.1'
+    awsVersion = '2.8.3'
+    mockitoVersion = '3.12.4'
+    robolectricVersion = '4.6.1'
+    retrofitVersion = '2.9.0'
+    materialVersion = '1.4.0'
+    ePoxyVersion = '4.6.3'
```
Comment (gpeal) [inline build.gradle:18]:
```suggestion
    epoxyVersion = '4.6.3'
```

File: gradle/base.gradle (hunk 2 above)
Comment (gpeal) [reply gradle/base.gradle:1]: Please revert them

File: sample-compose/build.gradle [hunk 4]
```diff
@@ -1,23 +1,18 @@
 plugins {
   id 'com.android.application'
-  id 'kotlin-android'
+  id "kotlin-android"
```
Comment (gpeal) [inline sample-compose/build.gradle:3]: Any reason you changed just this one?

**Review by gpeal — APPROVED** [review]

File: sample-compose/build.gradle (hunk 4 above)
Comment (Goooler) [reply sample-compose/build.gradle:3]: No, just effected by reverting.
